eXist-db 2.2 web.xml validation issue

Not sure if there are plans for maintenance releases in 2.x. But there is a validation issue with web.xml and it'd be great to fix it. There is a big stacktrace which ends with:

```
Caused by: javax.xml.stream.XMLStreamException: ParseError at [row,col]:[350,7]
Message: Unexpected element '{http://java.sun.com/xml/ns/j2ee}init-param' encountered
```

I face it when building war and deploying it under JBoss AS 7. Element `init-param` is not allowed as a child of `servlet-mapping` and should be added under `servlet` element instead. The issue occurs for `ResourceServlet`. The valid code will look like:

```
<servlet>
   <servlet-name>ResourceServlet</servlet-name>
   <servlet-class>
      de.betterform.agent.web.resources.ResourceServlet
   </servlet-class>
   <init-param>
      <param-name>caching</param-name>
      <param-value>true</param-value>
   </init-param>
</servlet>
...
<servlet-mapping>
   <servlet-name>ResourceServlet</servlet-name>
   <url-pattern>/bfResources/*</url-pattern>
</servlet-mapping>
```

--- a/extensions/betterform/main/etc/MergeWebXML.xsl
+++ b/extensions/betterform/main/etc/MergeWebXML.xsl
@@ -41,9 +41,7 @@
 </xsl:text>
         <servlet>
             <servlet-name>Flux</servlet-name>
-            <servlet-class>
-                org.directwebremoting.servlet.DwrServlet
-            </servlet-class>
+            <servlet-class>org.directwebremoting.servlet.DwrServlet</servlet-class>
             <init-param>
                 <param-name>debug</param-name>
                 <param-value>true</param-value>
@@ -51,27 +49,23 @@
         </servlet>
         <servlet>
             <servlet-name>XFormsPostServlet</servlet-name>
-            <servlet-class>
-                de.betterform.agent.web.servlet.XFormsPostServlet
-            </servlet-class>
+            <servlet-class>de.betterform.agent.web.servlet.XFormsPostServlet</servlet-class>
         </servlet>
         <servlet>
             <servlet-name>FormsServlet</servlet-name>
-            <servlet-class>
-                de.betterform.agent.web.servlet.FormsServlet
-            </servlet-class>
+            <servlet-class>de.betterform.agent.web.servlet.FormsServlet</servlet-class>
         </servlet>
         <servlet>
             <servlet-name>inspector</servlet-name>
-            <servlet-class>
-                de.betterform.agent.web.servlet.XFormsInspectorServlet
-            </servlet-class>
+            <servlet-class>de.betterform.agent.web.servlet.XFormsInspectorServlet</servlet-class>
         </servlet>
         <servlet>
             <servlet-name>ResourceServlet</servlet-name>
-            <servlet-class>
-                de.betterform.agent.web.resources.ResourceServlet
-            </servlet-class>
+            <servlet-class>de.betterform.agent.web.resources.ResourceServlet</servlet-class>
+            <init-param>
+                <param-name>caching</param-name>
+                <param-value>true</param-value>
+            </init-param>
         </servlet>
         <servlet>
             <servlet-name>error</servlet-name>
@@ -106,10 +100,6 @@
         <servlet-mapping>
             <servlet-name>ResourceServlet</servlet-name>
             <url-pattern>/bfResources/*</url-pattern>
-            <init-param>
-                <param-name>caching</param-name>
-                <param-value>true</param-value>
-            </init-param>
         </servlet-mapping>
         <servlet-mapping>
             <servlet-name>error</servlet-name>
@@ -123,9 +113,7 @@
 </xsl:text>
         <filter>
             <filter-name>XFormsFilter</filter-name>
-            <filter-class>
-                de.betterform.agent.web.filter.XFormsFilter
-            </filter-class>
+            <filter-class>de.betterform.agent.web.filter.XFormsFilter</filter-class>
         </filter>
         <xsl:text>
 </xsl:text>

